lib/bolt/target.rb

```diff
@@ -80,6 +80,17 @@ def to_h
       )
     end
 
+    def target_config_hash
+      {
```

Comment:
I think we should print the entire data structure for the target rather then the "compacted config"

so something like
```
{ name: foo
  uri: foo@example.com
  config: { transport: ssh, ssh: { }, winrm{ } }.
  vars: { } 
  etc
}
```

Reply:
Looking at the target data structure, it would make sense to include `name`, `uri`, `alias`, `config`, `vars`, `facts`, and `features`. There's also `plugin_hooks`, though I'm not sure if that should be included?

For fields that are emtpy (i.e. a target doesn't have a `uri`) do we want to remove that from the displayed data, or show the user that it's empty/unset?

Reply:
I think we want to show that it's empty or unset. I think plugin_hooks is also useful for debugging -  I can slightly see how it could be confusing if people think it's plugins that generated the target or something? But that seems unlikely.